bpfassets/libbpf/src/kepler.bpf.c

```diff
@@ -196,6 +196,7 @@ static inline void collect_metrics_and_reset_counters(
 
 struct task_struct {
 	int pid;
+	unsigned int __state;
```

Comment:
Is it fixed / defined what the size of  `int` is ?

Reply:
Yes. https://github.com/sustainable-computing-io/kepler/blob/main/bpfassets/libbpf/src/kepler.bpf.h#L8

Reply:
Also, these fields are BTF relocated - which will perform the necessary checks around int size and signedness at program load time.

Reply:
Thanks Dave. 
Related question ...  shouldn't we use `unsigned int` or `u32`  for `pid` as well?

Reply:
Shouldn't we use BPF_CORE_READ() to read the pointers in a safe manner?

Reply:
@sthaha I'm just using whatever is used in `task_struct` in the kernel - see https://elixir.bootlin.com/linux/latest/source/include/linux/sched.h#L977

if you follow the links from `pid_t` down to a concrete type you get to https://elixir.bootlin.com/linux/latest/source/include/uapi/asm-generic/posix_types.h#L28

@marceloamaral no you don't need to use BPF_CORE_READ in tp_btf programs.
https://mozillazg.com/2022/06/ebpf-libbpf-btf-powered-enabled-raw-tracepoint-common-questions-en.html#hidthe-difference-between-btf-raw-tracepoint-and-raw-tracepoint

Reply:
Thanks a lot for clarifications, @dave-tucker 🙏